HPCC-21249 Correct host.py to exactly match the localhost IP.

gfortil opened the pull request:
…-21249

Signed-off-by: Godson Fortil <FortGo01@risk.regn.net>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [x] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->
This change has been tested on a 104 node cluster and works as expected.
<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

gaojx committed:
HPCC-20780 Change the return value of encodeXML(), decodeXML() to be more rational

richardkchapman committed:
Split of 7.2.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Upmerge 7.0.2 changes

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Upmerge 7.2.0 Changes

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-20871 Avoid automatic type conversion via IpAddress ctor

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

Commit:
HPCC-20869 Convert Roxie XSD to new format

Convert XSD to new format

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

richardkchapman committed:
Merge branch 'candidate-7.0.2' into candidate-7.2.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.2.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

lpezet committed:
HPCC-20900 ISO implementation of week dates with tests.

Fixed ISO week dates functions.

Fixed Module name.

Implementation fixed. More tests added.

Formatting fixes.

Replaced tabs with spaces for new tests.

Comment fixed. Spaces fixed.

Spaces fixed.

Spaces fixed.

Using INTEGER2 for year now.

ghalliday committed:
Merge pull request #11882 from jakesmith/hpcc-20871

HPCC-20871 Avoid automatic type conversion via IpAddress ctor

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

wangkx committed:
HPCC-21339 Report thor job queue from ThorQueueMonitor

A new WsWorkunits method WUGetThorJobQueue is added to report
the information about thor job queue from ThorQueueMonitor.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

ghalliday committed:
HPCC-20856 Mark conditional errors as non-throwing

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-20978 Ensure ,DISTRIBUTE creates a distributed dataset when possible

Also adds support for ,DISTRIBUTED for some other forms of inline dataset

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

BrianB644 committed:
HPCC-18956: Adjust eclcc to emit valid code when compiling for Linux/ARM using GCC

Signed-off-by: BrianB644 <brianb@brianb.com>

richardkchapman committed:
HPCC-21029 Improve AsyncFor code

Executes the last cycle on the calling thread, in the common simple case.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

BrianB644 committed:
HPCC-18956: Adjust eclcc to emit valid code when compiling for Linux/ARM using GCC

Signed-off-by: BrianB644 <brianb@brianb.com>

RussWhitehead committed:
HPCC-21090 Extend default LDAP cache timeout

Reloading the LDAP security cache can take a while since there are quite
a few hpccinternal and other managed scopes. Currently the default for
a cache refresh is 5 minutes, which is way too often. This PR changes the
default for cache reload to every hour.

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

Reloading the LDAP security cache can take a while since there are quite
a few hpccinternal and other managed scopes. Currently the default for
a cache refresh is 5 minutes, which is way too often. This PR changes the
default for cache reload to every hour.

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

Commit:
HPCC-20947 Convert logging agent XSDs to new format

Convert ESP and Cassandra logging agent XSDs to new format

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

Commit:
HPCC-10949 Convert logging manager XSD to new format

Convert XSD to new format

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

Commit:
HPCC-20955 Convert Topology and related XSDs to new format

Convert XSDs to new format

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

davidarcher committed:
Fix dev doc links in README

davidarcher committed:
Update README.md

davidarcher committed:
Update README.md

davidarcher committed:
Fix memory manager link

richardkchapman committed:
HPCC-21018 Use persistent threads for parallel child queries in Roxie

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-21038 Issues building OSX with _USE_TBB set

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-21041 Reimplement the code to detect ambiguous selectors

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
HPCC-20980 Automatically build and distribute jar files for embedded Java code

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-21274 Allow jar files required by a workunit to be passed in manifest

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-21041 Delete code that built up a list of hidden tables

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
Merge pull request #11895 from ghalliday/issue20856

HPCC-20856 Mark conditional errors as non-throwing

Reviewed-By: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #11923 from davidarcher/patch-1

Fix doc links in README

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #11937 from richardkchapman/hpcc-21038

HPCC-21038 Issues building OSX with _USE_TBB set

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #11913 from BrianB644/HPCC-18956

HPCC-18956: Adjust eclcc to emit valid code when compiling for Linux/ARM using GCC

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #11917 from kenrowland/HPCC-20947

HPCC-20947 Convert logging agent XSDs to new format

Reviewed-By: Kevin Wang <kevin.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.0.4' into candidate-7.2.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.2.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'master' into candidate-7.2.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-21039 Add a variant of CPersistentThreaded that uses tbb tasks

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #11840 from gaojx/HPCC-20780_rationalize-return-of-encodeXML-decodeXML

HPCC-20780 Change the return value of encodeXML(), decodeXML() to be more rational

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #11936 from richardkchapman/hpcc-21018

HPCC-21018 Use persistent threads for parallel child queries in Roxie

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #11919 from kenrowland/HPCC-20949

HPCC-20949 Convert logging manager XSD to new format

Reviewed-By: Kevin Wang <kevin.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #11935 from richardkchapman/hpcc-21029

HPCC-21029 Improve AsyncFor code

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
HPCC-21062 Add ECL syntax for DISTRIBUTE(ds, ALL)

Also implement much of the underlying code to support

    DISTRIBUTE(ds, f(LEFT, TARGETNODE))

to allow distribution to a set of nodes

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
Merge pull request #11942 from richardkchapman/persistentTask

HPCC-21039 Add a variant of CPersistentThreaded that uses tbb tasks

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.0.4' into candidate-7.2.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.2.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #11939 from richardkchapman/embed-java

HPCC-20980 Automatically build and distribute class files for embedded Java code

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
HPCC-20933 Fix the interpretation of an active table for no_selectnth

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

jakesmith committed:
HPCC-21077 Add a statistic for graph execution count

Track and publish number of times each graph executes.

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

Commit:
HPCC-20917 Convert WsECL ESP Service XSD to new format

Deprecate old WsECL XSD
Convert current XSD in use for WsECL ESP Service to new format

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

Commit:
HPCC-21080 Conversion of LDAP Server process XSD

Convert XSD to new format

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

richardkchapman committed:
HPCC-21142 Add persistence options to Java embed

Also fixes:

HPCC-21052 Use thread local variables for IEmbedFunctionContext
HPCC-21137 Problems with embedded java tests on master
HPCC-21127 Incorrect name mangling rules (additional fix)

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #11940 from ghalliday/issue20978

HPCC-20978 Ensure ,DISTRIBUTE creates a distributed dataset when possible

Reviewed-By: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-20961 Optimize removing a memory callback

It is often the case that a memory callback of the same cost will be
registered in the near future - so it is better to retain the entry
in the callback lists, rather than repeatedly adding and removing.

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
Merge branch 'candidate-7.0.4' into candidate-7.2.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.2.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

kunalaswani committed:
HPCC-20815 Add expireDays attribute to Logical Files

Add the expireDays attribute to the Logical Files.

Signed-off-by: Aswani, Kunal <kunal.aswani@lexisnexis.com>

richardkchapman committed:
Merge pull request #11944 from kenrowland/HPCC-20955

HPCC-20955 Convert Topology and related XSDs to new format

Reviewed-By: Russ Whitehead <william.whitehead@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #11943 from ghalliday/issue21049

HPCC-21062 Add ECL syntax for DISTRIBUTE(ds, ALL)

Reviewed-By: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #11955 from ghalliday/issue20961

HPCC-20961 Optimize removing a memory callback

Reviewed-By: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

Commit:
HPCC-21087 Conversion of Directories XSD to new format

Convert XSD to new format

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

richardkchapman committed:
Merge pull request #11953 from kenrowland/HPCC-21080

HPCC-21080 Conversion of LDAP Server process XSD

Reviewed-By: Russ Whitehead <william.whitehead@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #11951 from kenrowland/HPCC-20917

HPCC-20917 Convert WsECL ESP Service XSD to new format

Reviewed-By: Russ Whitehead <william.whitehead@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

Commit:
HPCC-21124 Convert WS SQL XSD to new format

Conver XSD to new format

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

richardkchapman committed:
Merge pull request #11938 from ghalliday/issue21041

 HPCC-21041 Reimplement the code to detect ambiguous selectors

Reviewed-By: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-21127 Incorrect name mangling rules

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-21128 Compile-time syntax check for python plugin

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

Commit:
HPCC-18969 Implement wizard support for new config manager

Add support for templated configuration changes

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

Comment out all code, just headers

jakesmith committed:
HPCC-21049 Thor implementation of DISTRIBUTE,ALL

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

ghalliday committed:
Merge pull request #11971 from richardkchapman/hpcc-21127

HPCC-21127 Incorrect name mangling rules

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

Commit:
HPCC-21134 Convert WS Logging XSD to new format

Convert XSD to new format

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

Commit:
HPCC-21136 Convert DropZone XSD to new format

Convert DropZone XSD to new format
Update CMakeLists.txt for new XSDs to automatically process all XSDs,
including build time substitutions.

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

richardkchapman committed:
Merge pull request #11970 from kenrowland/HPCC-21124

HPCC-21124 Convert WS SQL XSD to new format

Reviewed-By: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #11967 from kenrowland/HPCC-18969

HPCC-18969 Implement wizard support for new config manager

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #11972 from richardkchapman/syntax-python

 HPCC-21128 Compile-time syntax check for python plugin

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #11949 from jakesmith/hpcc-21077

HPCC-21077 Add a statistic for graph execution count

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #11962 from kenrowland/HPCC-21087

HPCC-21087 Conversion of Directories XSD to new format

Reviewed-By: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-21131 Fix minor memory leak for DISTRIBUTE,ALL

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

jakesmith committed:
HPCC-21147 Avoid linking issue exposed by de-virtualization

ws_espcontrol calls a virtual in CEspServer, which in turn
calls some code in CEspConfig.
With de-virtualization, the compiler might inline the virtual,
when it does that, the CEspConfig method is not linked to the
ws_espcontrol library and an unresolved symbol is fired at
runtime.

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

jakesmith committed:
HPCC-21146 Ensure implicitkeydistribute.ecl has a consistent order

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

richardkchapman committed:
HPCC-21142 Add persistence options to Java embed

Additional fixes in this area, plus refactored code..

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #11982 from jakesmith/hpcc-21147

HPCC-21147 Avoid linking issue exposed by de-virtualization

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #11983 from jakesmith/hpcc-21146

HPCC-21146 Ensure implicitkeydistribute.ecl has a consistent order

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #11980 from ghalliday/issue21131

HPCC-21131 Fix minor memory leak for DISTRIBUTE,ALL

Reviewed-By: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #11974 from richardkchapman/java-persist

HPCC-21142 Add persistence options to Java embed

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

dcamper committed:
HPCC-20951 Add Std.Date.FromStringToSeconds() function

Parses strings containing both date and time values (e.g. '1999-02-01T12:34:56') and converts the results into a Std.Date.Seconds_t value (number of seconds since epoch).

Signed-off-by: Dan S. Camper <dan.camper@lexisnexisrisk.com>

Fix whitespaces

Signed-off-by: Dan S. Camper <dan.camper@lexisnexisrisk.com>

Changes based on code review

Signed-off-by: Dan S. Camper <dan.camper@lexisnexisrisk.com>

Changes based on code review

Signed-off-by: Dan S. Camper <dan.camper@lexisnexisrisk.com>

Changes for code review

Signed-off-by: Dan S. Camper <dan.camper@lexisnexisrisk.com>

richardkchapman committed:
Merge branch 'candidate-7.0.6' into candidate-7.2.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.2.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #11941 from jakesmith/hpcc-21049

HPCC-21049 Thor implementation of DISTRIBUTE,ALL

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #11973 from RussWhitehead/cache720

HPCC-21090 Extend default LDAP cache timeout

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

Commit:
HPCC-21089 Add user descriptor serialize without password method

Currently if you want to send a user descriptor without a password, you have to
create a new one, copy the old one without the password, and serialize it. This
PR adds a new "serializeWithoutPassword" methodr

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexisrisk.com>

Commit:
HPCC-21121 Maintain XSD attribute and element order as set in XSDs

Add ordinal value to attribute for use by UI
Store elements in vector to maintain order

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

ghalliday committed:
HPCC-21169 Better processing of assignment to fields within rows

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
Merge pull request #11984 from ghalliday/issue20933b

HPCC-20933 Fix the interpretation of an active table for no_selectnth

Reviewed-By: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #11976 from kenrowland/HPCC-21136

HPCC-21136 Convert DropZone XSD and auto install all XSDs in CMakeLists.txt

Reviewed-By: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #11965 from kenrowland/HPCC-21121

HPCC-21121 Maintain XSD attribute and element order as set in XSDs

Reviewed-By: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #11975 from kenrowland/HPCC-21134

HPCC-21134 Convert WS Logging XSD to new format

Reviewed-By: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'hpcc-20900' into candidate-7.2.0

Reviewed-By: Dan Camper <dan.camper@lexisnexis.com>
Reviewed-By: Richard
Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.0.6' into candidate-7.2.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.2.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-21199 Embedded JS and Python test cases failed in OBT

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #11998 from ghalliday/issue21169

HPCC-21169 Better processing of assignment to fields within rows

Reviewed-By: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #11992 from RussWhitehead/noPwd

HPCC-21089 Add bool flag to user descriptor serialize password

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #11999 from richardkchapman/hpcc-21199

HPCC-21199 Embedded JS test cases failed in OBT

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

jakesmith committed:
HPCC-21215 Detect CSV/XML/FLAT/INDEX mismatches.

Only throw exception in case of variable diskread reading
mismatched type, or flat vs index (as before).
Otherwise issue workunit warning only.

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

ghalliday committed:
Merge pull request #11988 from richardkchapman/java-persist-experiment

HPCC-21142 Add persistence options to Java embed

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.0.6' into candidate-7.2.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.0.6' into candidate-7.2.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.2.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-21222 Datasets passed to R dataframes should use Vectors rather than lists

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #11989 from dcamper/hpcc-20951-datetime-diff

HPCC-20951 Add Std.Date.FromStringToSeconds() function

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
HPCC-21223 Roxie may terminate on exceptions in 7.2 branch

Can't throw exceptions out of a destructor

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-21224 Compile-time syntax check for R plugin

Also fixes a lurking bug in the demangler used when constant-folding external
functions.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-21216 #EXPORT should not remove digits from the type for records

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
HPCC-21228 Support passing java objects to embedded code

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

shamser committed:
HPCC-21229 When generating ECL, only lower case id's if lowerCaseIds(true)

Signed-off-by: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>

ghalliday committed:
Merge pull request #12011 from richardkchapman/R-syntax

HPCC-21224 Compile-time syntax check for R plugin

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #12015 from shamser/issue21229

HPCC-21229 When generating ECL, only lower case id's if lowerCaseIds(true)

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #12014 from richardkchapman/java-object

HPCC-21228 Support passing java objects to embedded code

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.0.6' into candidate-7.2.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.2.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #12006 from jakesmith/hpcc-21215

HPCC-21215 Detect CSV/XML/FLAT/INDEX mismatches.

Reviewed-By: Gavin Halliday <gavin.halliday@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #12013 from ghalliday/issue21216

HPCC-21216 #EXPORT should not remove digits from the type for records

Reviewed-By: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #12010 from richardkchapman/HPCC-21223

HPCC-21223 Roxie may terminate on exceptions in 7.2 branch

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #12008 from richardkchapman/hpcc-21222

HPCC-21222 Datasets passed to R dataframes should use Vectors rather than lists

richardkchapman committed:
Merge branch 'candidate-7.0.8' into candidate-7.2.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.2.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-21256 CLASSPATH option on java import not working

Regression in 7.2.0 branch

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-21257 Classes loaded using same classpath should be shared

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.0.8' into candidate-7.2.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.2.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #12028 from richardkchapman/hpcc-21256

HPCC-21256 CLASSPATH option on java import not working

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.0.8' into candidate-7.2.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.2.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-21271 Some regression test suites failing due to incorrect keys

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-21270 Incorrect temporary directory used in make install mode

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #12044 from richardkchapman/hpcc-21270

HPCC-21270 Incorrect temporary directory used in make install mode

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
HPCC-21277 Improve exception reporting from java embed

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #12031 from richardkchapman/java-local-jar

HPCC-21257 Classes loaded using same classpath should be shared 

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #12051 from richardkchapman/java-exception

HPCC-21277 Improve exception reporting from java embed

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
HPCC-21283 Add .java files to build process rather than checking in classes

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #12042 from richardkchapman/hpcc-21271

HPCC-21271 Some regression test suites failing due to incorrect keys

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.0.8' into candidate-7.2.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.2.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

Commit:
HPCC-20458 Allow File Access keys to be defined in the environment

Add new CLI envmod for processing modification templates to modify
environment files.
Update modification template support.
Add unit tests for new features.

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

richardkchapman committed:
Merge pull request #11957 from kunalaswani/HPCC-20815

HPCC-20815 Add expireDays attribute to Logical Files

Reviewed-By: Miguel Vazquez <miguel.vazquez@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #12039 from richardkchapman/manifest-jar

HPCC-21274 Allow jar files required by a workunit to be passed in manifest

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

Commit:
HPCC-21303 Convert dynamic ESDL XSD to new format

Convert dynamic ESDL XSD to new Config Manager 2.0 format
Remove unnecessary attributes from existing XSDs
Fix a few incorrect XSD elements

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

Commit:
HPCC-21292 Convert Sasha XSD to new format

Convert Sasha XSD to new Config Manager XSD format

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

richardkchapman committed:
Merge pull request #12063 from kenrowland/HPCC-21303

HPCC-21303 Convert dynamic ESDL XSD to new format

Reviewed-By: James McMullan <james.mcmullan@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #12027 from kenrowland/HPCC-20458

HPCC-20458 Allow File Access keys to be defined in the environment

Reviewed-By: James McMullan <james.mcmullan@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.0.8' into candidate-7.2.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.2.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #12062 from kenrowland/HPCC-21292

HPCC-21292 Convert Sasha XSD to new format

Reviewed-By: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-21317 Update cassandra drive submodule to latest

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-21316 Build break with boost version 57

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

Commit:
HPCC-21311 Convert Daliplugin XSD to new format

Convert Daliplugin XSD to new config manager 2 format

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

Commit:
HPCC-21319 Convert mysqlserver XSD to new format

Convert XSD to new Config Manager 2.0 format

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

gfortil commented:
@xwang2713 please review.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 2587a0902dfcce8290342b56cf0b41c6b986e1a0
Build: success
Build: success
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |101 |101 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |812 |812 |0 |
|test (thor) |738 |738 |0 |
|test (roxie) |886 |886 |0 |


HPCC Stop: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   31 sec (00:00:31) |  190 sec (00:03:10) |    0 sec (00:00:00) |    2 sec (00:00:02) |   18 sec (00:00:18) | 1413 sec (00:23:33) |   19 sec (00:00:19) | 1673 sec (00:27:53) |

ghalliday committed:
Merge pull request #12069 from richardkchapman/osx-break

HPCC-21316 Build break with boost version 57

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #12068 from richardkchapman/new-cassandra

HPCC-21317 Update cassandra drive submodule to latest

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #12053 from richardkchapman/java-build

HPCC-21283 Add .java files to build process rather than checking in classes

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.0.8' into candidate-7.2.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.2.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-20458 Fix build break on enterprise build

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-21325 Improve roxiemem huge allocation stress tests

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
Merge branch 'candidate-7.0.8' into candidate-7.2.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.2.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #12081 from ghalliday/issue21325

HPCC-21325 Improve roxiemem huge allocation stress tests

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #12079 from ghalliday/issue20458fix

HPCC-20458 Fix build break on enterprise build

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

dcamper committed:
HPCC-21337 Fix JavaCat source/target version

Explicitly specify a source version of 1.8

Signed-off-by: Dan S. Camper <dan.camper@lexisnexisrisk.com>

richardkchapman committed:
Merge pull request #12088 from dcamper/hpcc-21337-javac-source

HPCC-21337 Fix JavaCat source/target version

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-21337 Fix javaembed build error (source/target version)

Previous fix mised one instance that needed -source.

Also remove registred TM symbol as it caused Java some issues.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #12092 from richardkchapman/hpcc-21337

HPCC-21337 Fix javaembed build error (source/target version)

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #12070 from kenrowland/HPCC-21311

HPCC-21311 Convert Daliplugin XSD to new format
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #12072 from kenrowland/HPCC-21319

HPCC-21319 Convert mysqlserver XSD to new format

Reviewed-By: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #12091 from wangkx/h21339

HPCC-21339 Report thor job queue from ThorQueueMonitor

Reviewed-By: Russ Whitehead <william.whitehead@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.0.10' into candidate-7.2.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-7.2.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

HPCCSmoketest commented:
Process of PR-12074, label: master starts now. Estimated completion time is ~ 0.5 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 7e1d50a90a3119c1cc7706830e2cc1e365f0e7d0
Build: success
Build: success
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |101 |101 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |812 |812 |0 |
|test (thor) |738 |738 |0 |
|test (roxie) |886 |886 |0 |


HPCC Stop: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   28 sec (00:00:28) |  195 sec (00:03:15) |    0 sec (00:00:00) |    2 sec (00:00:02) |   19 sec (00:00:19) | 1606 sec (00:26:46) |   19 sec (00:00:19) | 1869 sec (00:31:09) |

xwang2713 reviewed (approved):
Looks good

richardkchapman commented:
Should this be targeting 7.0.12 ?

gfortil commented:
I believe that would be fine. @xwang2713 could you please confirm?

richardkchapman commented:
Then you will need to rebase this change --onto candidate-7.0.12, force-push it, then edit the target branch in this PR @gfortil

HPCCSmoketest commented:
Process of PR-12074, label: master starts now. Estimated completion time is ~ 0.5 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 7e1d50a90a3119c1cc7706830e2cc1e365f0e7d0
Number of Error(s): 1
-- Configuring incomplete, errors occurred!
Number of Error(s): 1
-- Configuring incomplete, errors occurred!
CMakeFiles/CMakeOutput.log copied into ../CMakeOutput-2019-01-28_09-23-51.log

[Build error:](https://gist.github.com/b1d383c3468161bda78328022c7a992d)
Build log [Build log](https://gist.github.com/HPCCSmoketest/b1d383c3468161bda78328022c7a992d#file-result-19-01-28-09-23-28-log)

HPCCSmoketest commented:
Process of PR-12074, label: master starts now. Estimated completion time is ~ 0.5 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: b10df09cc2aa8e6f7cfa1849fa5ba1dfb71a668a
Number of Error(s): 1
-- Configuring incomplete, errors occurred!
Number of Error(s): 1
-- Configuring incomplete, errors occurred!
CMakeFiles/CMakeOutput.log copied into ../CMakeOutput-2019-01-28_11-57-01.log

[Build error:](https://gist.github.com/bd2e42981565385f90594f703878c520)
Build log [Build log](https://gist.github.com/HPCCSmoketest/bd2e42981565385f90594f703878c520#file-result-19-01-28-11-56-45-log)

HPCCSmoketest commented:
Process of PR-12074, label: master starts now. Estimated completion time is ~ 0.5 hour(s)

Commit:
HPCC-21249 Correct host.py to exactly match the localhost IP.

Signed-off-by: Godson Fortil <FortGo01@risk.regn.net>

HPCCSmoketest commented:
Process of PR-12074, label: master starts now. Estimated completion time is ~ 0.5 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: dbfb6edb9b1190417629e3311e05e9c8095568fe
Number of Error(s): 1
-- Configuring incomplete, errors occurred!
Number of Error(s): 1
-- Configuring incomplete, errors occurred!
CMakeFiles/CMakeOutput.log copied into ../CMakeOutput-2019-01-28_14-51-34.log

[Build error:](https://gist.github.com/8853806abf418f34adb49f510ff0e3d7)
Build log [Build log](https://gist.github.com/HPCCSmoketest/8853806abf418f34adb49f510ff0e3d7#file-result-19-01-28-14-51-23-log)

xwang2713 reviewed (approved):
Looks good

richardkchapman commented:
@gfortil Your rebase did not succeed
